runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/config/KafkaTopicConfigAdapter.java

```diff
@@ -96,7 +97,12 @@ public KafkaTopicConfig adaptFromJson(
 
         if (key != null)
         {
-            keyConfig = validator.adaptFromJson(key);
+            JsonObjectBuilder keyObject = Json.createObjectBuilder();
+
+            key.forEach(keyObject::add);
+            keyObject.add(SUBJECT, name + "-key");
```

Comment:
This is a helpful way to handle it, appending the correct suffix in the context of key vs value.
However, this is no longer round-tripping correctly after adapting to and from JSON.
Instead, let's apply the suffix where the value is used, and keep the parsing code generic, reading and writing the values verbatim.

Reply:
After digging into this a bit more, I realized the fix has deeper impact than we need at the moment, so we can go with this approach for now.